src/plugins/data/public/ui/dataset_selector/configurator.tsx

```diff
@@ -112,15 +137,16 @@ export const Configurator = ({
                     text: field.displayName || field.name,
                     value: field.name,
                   })),
-                  { text: '-----', value: '', disabled: true },
-                  { text: 'No time field', value: undefined },
+                  { text: '-----', value: '-----', disabled: true },
+                  { text: noTimeFilter, value: noTimeFilter },
                 ]}
                 value={timeFieldName}
                 onChange={(e) => {
-                  const value = e.target.value === 'undefined' ? undefined : e.target.value;
-                  setTimeFieldName(value);
+                  const value = e.target.value === noTimeFilter ? undefined : e.target.value;
+                  setTimeFieldName(e.target.value);
                   setDataset({ ...dataset, timeFieldName: value });
                 }}
+                hasNoInitialSelection={dataset.type !== DEFAULT_DATA.SET_TYPES.INDEX_PATTERN}
```

Comment:
Do we still need this with the logic i proposed?

Reply:
Nope, should default to true then. Updated in 51e4490